Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/Sync/ChangeEvent.swift

```diff
@@ -44,12 +53,54 @@ public struct ChangeEvent<DocumentT: Codable>: Codable, Hashable {
     /// Whether or not this ChangeEvent has pending writes.
     public let hasUncommittedWrites: Bool
 
+    init(id: AnyBSONValue,
+         operationType: OperationType,
+         fullDocument: DocumentT?,
+         ns: MongoNamespace,
+         documentKey: Document,
+         updateDescription: UpdateDescription?,
+         hasUncommittedWrites: Bool) {
+        self.id = id
+        self.operationType = operationType
+        self.fullDocument = fullDocument
+        self.ns = ns
+        self.documentKey = documentKey
+        self.updateDescription = updateDescription
+        self.hasUncommittedWrites = hasUncommittedWrites
+    }
+
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+
+        self.id = try container.decode(AnyBSONValue.self, forKey: .id)
+        self.operationType = OperationType.init(
+            rawValue: try container.decode(String.self, forKey: .operationType)
+        ) ?? .unknown
```

Comment:
[minor] should be `try?`

Reply:
If the operationType key should always be there, this should fail if the key doesn't exist.

Reply:
Oh I see, this would only fail if `container.decode` fail if it's not a string. In that case that's fine